delta/plugins/archive/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/archive/model/ArchiveReference.scala

```diff
@@ -74,4 +79,57 @@ object ArchiveReference {
       ref.original.toString + project.toString + path.toString + representation.toString
   }
 
+  implicit final val archiveReferenceJsonLdDecoder: JsonLdDecoder[ArchiveReference] = {
+    // custom decoder for resource ref to rev and tag references from separate fields
+    implicit val resourceRefDecoder: JsonLdDecoder[ResourceRef] =
+      (cursor: ExpandedJsonLdCursor) =>
+        for {
+          iri    <- cursor.downField(nxv + "resourceId").get[Iri]
```

Comment:
This should probably go in the companion object of `ResourceRef `

Reply:
it can't because it's specific to archives, it needs to deserialize variants like:
```
{
  "resourceId": "http://localhost/id"
}
```
```
{
  "resourceId": "http://localhost/id",
  "tag": "mytag"
}
```
```
{
  "resourceId": "http://localhost/id",
  "rev": 1
}
```
into a `ResourceRef`.

`ResourceRef` already has a decoder that follows the following formats:
```
"http://localhost/id"
"http://localhost/id?rev=1"
"http://localhost/id?tag=mytag"
```